src/solidity/stateDecoder.js

```diff
@@ -0,0 +1,233 @@
+        location = this.moveNextArray(location)
+      }
+    } else {
+      for (var i = 0; i < size; i++) {
+        ret.push(this.decodeType(type, storage, location))
+        location = this.moveNextItemInArray(type, location)
+      }
+    }
+    return ret
+  },
+
+  moveNextArray: function (location) {
+    return {
+      offset: 0,
+      slot: location.slot + 1
+    }
+  },
+
+  moveNextItemInArray: function (type, location) {
+    if (type.size === 'dynamic' || location.offset + type.memSize >= 64 || type.isStruct) {
+      location.offset = 0
+      location.slot = add(location.slot, 1)
+    } else {
+      location.offset = location.offset + type.memSize
+    }
+    return location
+  },
+
+  decodeInt: function (type, storage, location) {
+    var value = getValue(type, storage, location)
+    value = extractValue(value, type, location)
+    var bigNumber = new BN(value.replace('0x', ''), 16)
+    if (type.innerType === 'uint') {
+      return bigNumber.toString(10)
+    } else if (type.innerType === 'int') {
+      if (this.isNegative(value)) {
+        return '-' + bigNumber.sub(new BN(new Array(value.length + 1).join('f'), 16)).sub(new BN(1)).toString(10)
+      } else {
+        return bigNumber.toString(10)
+      }
+    }
+  },
+
+  decodeStruct: function (type, storage, location) {
+    var ret = {}
+    for (var k in type.members) {
+      var member = type.members[k]
+      location = {
+        slot: add(member.location.slot, location.slot),
+        offset: member.location.offset
+      }
+      ret[member.name] = this.decode(member.type, storage, location)
+    }
+    return ret
+  },
+
+  decodeBool: function (type, storage, location) {
+    var value = getValue(type, storage, location)
+    value = extractValue(value, type, location)
+    return value === '0x01'
+  },
+
+  decodeEnum: function (type, storage, location) {
+    var value = getValue(type, storage, location)
+    value = extractValue(value, type, location)
+    value = parseInt(value)
+    if (type.enum) {
+      return type.enum[value].attributes.name
+    }
+    return ''
+  },
+
+  decodeAddress: function (type, storage, location) {
+    var value = getValue(type, storage, location)
+    return extractValue(value, type, location)
+  },
+
+  decodeBytes: function (type, storage, location) {
+    if (type.size === 'dynamic') {
+      return this.decodeDynamicBytes(type, storage, location)
+    } else {
+      var value = getValue(type, storage, location)
+      return extractValue(value, type, location)
+    }
+  },
+
+  decodeDynamicBytes: function (type, storage, location) {
+    var value = getValue(type, storage, location)
+    var key = getDynamicPointer(location)
+    if (storage[key] && storage[key] !== '0x') {
+      var ret = ''
+      var length = parseInt(value) - 1
+      var slots = Math.ceil(length / 64)
+      var currentSlot = storage[key]
+      key = new BN(key.replace('0x', ''), 16)
```

Comment:
It seems that these `0x` prefixes are in the way at several points. Would it make sense to re-format the storage mapping and remove all of them?